std/uni.d

```diff
@@ -4711,7 +4717,7 @@ enum Mode {
 
 mixin template ForwardStrings()
 {
-    private bool fwdStr(string fn, C)(ref C[] str) const pure
+    private bool fwdStr(string fn, C)(ref C[] str) const @trusted
```

Comment:
Do we have a test for the purity here?

Reply:
Yes, with latest commit: Any utf\d*Matcher calling match/test/skip has it in it's call chain, e.g. pure @safe unittest beginning at line 4634 or 4670 or 5319 or 5374 or 5424. Added pure to many @safe unittest.
BTW, I failed introducing Mallocator in ReallocPolicy.alloc/realloc/destroy due to @system calls, like this:

@@ -1812,6 +1812,7 @@
 @safe struct ReallocPolicy
 {
     import std.range.primitives : hasLength;
+    import std.experimental.allocator.mallocator : Mallocator;
 
     static T[] dup(T)(const T[] arr)
     {
@@ -1820,9 +1821,10 @@
         return result;
     }
 
-    static T[] alloc(T)(size_t size) @trusted
+    static T[] alloc(T)(size_t size) //@trusted
     {
-        import core.stdc.stdlib : malloc;
+//        import core.memory : pureMalloc; //import core.stdc.stdlib : malloc;
+        import std.experimental.allocator : makeArray;
         import std.exception : enforce;
 
         import core.checkedint : mulu;
@@ -1830,13 +1832,14 @@
         size_t nbytes = mulu(size, T.sizeof, overflow);
         if (overflow) assert(0);
 
-        auto ptr = cast(T*) enforce(malloc(nbytes), "out of memory on C heap");
-        return ptr[0 .. size];
+        return enforce(Mallocator.instance.makeArray!ubyte(nbytes), "out of memory on C heap"); //auto ptr = cast(T*) enforce(pureMalloc(nbytes), "out of memory on C heap");
+//        return ptr[0 .. size];
     }